clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy

```diff
@@ -243,22 +244,26 @@ class OpenstackServerGroupCachingAgent extends AbstractOpenstackCachingAgent imp
     Map<String, Object> result = Maps.newHashMap()
 
     if (stack) {
-      result.put('minSize', stack.parameters?.get('min_size') ?: 0)
-      result.put('maxSize', stack.parameters?.get('max_size') ?: 0)
-      result.put('desiredSize', stack.parameters?.get('desired_size') ?: 0)
+      result.put('minSize', getInteger(stack.parameters.get('min_size')))
+      result.put('maxSize', getInteger(stack.parameters.get('max_size')))
+      result.put('desiredSize', getInteger(stack.parameters.get('desired_size')))
       result.put('autoscalingType', ServerGroupParameters.AutoscalingType.fromMeter(stack.parameters.get('autoscaling_type')))
       ['up','down'].each {
-        Map<String, Object> scaler = [cooldown  : stack.parameters.get("scale${it}_cooldown".toString()),
-                                      period    : stack.parameters.get("scale${it}_period".toString()),
-                                      adjustment: stack.parameters.get("scale${it}_adjustment".toString()),
-                                      threshold : stack.parameters.get("scale${it}_threshold".toString())]
+        Map<String, Object> scaler = [cooldown  : getInteger(stack.parameters.get("scale${it}_cooldown".toString()), null),
+                                      period    : getInteger(stack.parameters.get("scale${it}_period".toString()), null),
+                                      adjustment: getInteger(stack.parameters.get("scale${it}_adjustment".toString()), null),
+                                      threshold : getInteger(stack.parameters.get("scale${it}_threshold".toString()), null)]
         result.put("scale$it".toString(), scaler)
       }
     }
 
     result
   }
 
+  Integer getInteger(String value, Integer defaultValue = 0) {
```

Comment:
I didn't follow this so thoroughly, but seems like the same thing that this method is doing can be accomplished via `as Integer`.